src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.CompilationData.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    internal abstract partial class AnalyzerDriver : IDisposable
+    {
+        protected static readonly ConditionalWeakTable<Compilation, CompilationData> s_compilationDataCache = new ConditionalWeakTable<Compilation, CompilationData>();
```

Comment:
+1


-          Heejae

From: Paul Harrington [mailto:notifications@github.com]
Sent: Tuesday, September 8, 2015 1:03 PM
To: dotnet/roslyn <roslyn@noreply.github.com>
Cc: HeeJae Chang <Heejae.Chang@microsoft.com>
Subject: Re: [roslyn] Performance improvements for the v2 analyzer driver (#4988)


In src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.CompilationData.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2froslyn%2fpull%2f4988%23discussion_r38972689&data=01%7c01%7cHeejae.Chang%40microsoft.com%7ced2e5fd9ae4c4f874aad08d2b8888d5c%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=JYxWO1sxMQL4%2fujoT2%2fs4gDMYwNy%2fcZ8osJl4JVJvdE%3d>:

> @@ -0,0 +1,189 @@

> +﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.

> +

> +using System;

> +using System.Collections.Generic;

> +using System.Collections.Immutable;

> +using System.Runtime.CompilerServices;

> +using System.Threading;

> +

> +namespace Microsoft.CodeAnalysis.Diagnostics

> +{

> +    internal abstract partial class AnalyzerDriver : IDisposable

> +    {

> +        protected static readonly ConditionalWeakTable<Compilation, CompilationData> s_compilationDataCache = new ConditionalWeakTable<Compilation, CompilationData>();

Now would be a good time to add a general-purpose property bag to Compilation and move all the ConditionalWeakTable<Compliation, ...> instances we have into that.

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2froslyn%2fpull%2f4988%2ffiles%23r38972689&data=01%7c01%7cHeejae.Chang%40microsoft.com%7ced2e5fd9ae4c4f874aad08d2b8888d5c%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=o7clUpiveVamiM2fy7uc1S1Aih7rQlU5e2KWz7q42Ws%3d>.